openapi-cli/src/main/java/org/ballerinalang/openapi/cmd/OpenApiCmd.java

```diff
@@ -92,33 +93,43 @@
     public OpenApiCmd() {
         this.outStream = System.err;
         this.executionPath = Paths.get(System.getProperty("user.dir"));
+        this.exitWhenFinish = true;
     }
 
     public OpenApiCmd(PrintStream outStream, Path executionDir) {
         this.outStream = outStream;
         this.executionPath = executionDir;
+        this.exitWhenFinish = true;
     }
 
+    public OpenApiCmd(PrintStream outStream, Path executionDir, boolean exitWhenFinish) {
+        this.outStream = outStream;
+        this.executionPath = executionDir;
+        this.exitWhenFinish = exitWhenFinish;
+    }
     @Override
     public void execute() {
         //User notification of using an experimental tool
         outStream.println(OpenApiMesseges.EXPERIMENTAL_FEATURE);
         if (helpFlag) {
             String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(getName());
             outStream.println(commandUsageInfo);
+            exitError(this.exitWhenFinish);
             return;
         }
         //Check if cli input argument is present
         if (inputPath) {
             //Check if an OpenApi definition is provided
             if (argList == null) {
-                throw LauncherUtils.createLauncherException(OpenApiMesseges.MESSAGE_FOR_MISSING_INPUT);
+                outStream.println(OpenApiMesseges.MESSAGE_FOR_MISSING_INPUT);
+                exitError(this.exitWhenFinish);
+                return;
```

Comment:
Check if this `return` is needed.

Reply:
This return needs to terminate the command in the test cases